src/ark/analysis/neighborhood_analysis.py

```diff
@@ -509,3 +510,102 @@ def compute_mixing_score(fov_neighbors_mat, fov, target_cells, reference_cells,
         mixing_score = reference_target / (target_target + reference_reference)
 
     return mixing_score
+
+
+def shannon_diversity(proportions):
+    """ Calculates the shannon diversity index for the provided proportions of a community
+    Args:
+        proportions (np.array):
+            the proportions of each individual group
+
+    Returns:
+        float:
+            the diversity of neighborhood
+    """
+
+    prop_index = proportions > 0
+    return -np.sum(proportions[prop_index] * np.log2(proportions[prop_index]))
```

Comment:
What happens for cells with no neighbors?

Reply:
The neighbors matrix only includes cells with at least one neighbor. https://github.com/angelolab/ark-analysis/blob/abc15e7a01688f07b4a9331f5156164ff83e6962/src/ark/analysis/neighborhood_analysis.py#L98